lib/ddtrace/transport/http.rb

```diff
@@ -82,28 +80,6 @@ def default_adapter
         :net_http
       end
 
-      def default_hostname
-        return default_url.hostname if default_url
-
-        ENV.fetch(Datadog::Ext::Transport::HTTP::ENV_DEFAULT_HOST, Datadog::Ext::Transport::HTTP::DEFAULT_HOST)
-      end
-
-      def default_port
-        return default_url.port if default_url
-
-        ENV.fetch(Datadog::Ext::Transport::HTTP::ENV_DEFAULT_PORT, Datadog::Ext::Transport::HTTP::DEFAULT_PORT).to_i
-      end
-
-      def default_url
-        url_env = ENV.fetch(Datadog::Ext::Transport::HTTP::ENV_DEFAULT_URL, nil)
-
-        if url_env
-          uri_parsed = URI.parse(url_env)
-
-          uri_parsed if %w[http https].include?(uri_parsed.scheme)
-        end
-      end
```

Comment:
These are technically public APIs that we have in the past recommended users to access in their configuration blocks. A hard removal of these will break user applications today.

Would it be possible to delegate these to `agent_settings`, and emit a deprecation warning with alternatives instead of removing it right away?

Reply:
😭  🆗  -- changed in 68bd40195d1d82c75105e40e42d9df06d9952cf0. I could not delegate to `agent_settings` since this is a module function and there's no `agent_settings` around when those methods are called, but I used my favorite crutch for this PR the `ENVIRONMENT_AGENT_SETTINGS` to get the original behavior where we honored the environment variables.

One more reason why the `c.tracer.transport_options` needs to go away.